### ISSUE TITLE
"undefined entity" exception when parsing XHTML containing entities

### ISSUE THREAD

**Issue by cdauth** [jsdom/jsdom#3266]
I am trying to parse an XHTML document that is using HTML entities. Parsing the document as XHTML rather than HTML is a requirement because the document contains CDATA sections, which are not interpreted correctly in HTML mode. In the browser (Chromium), I can use `DOMParser` in XML mode as long as I declare an XHTML DOCTYPE. In jsdom, an exception is thrown about the entity, regardless of the DOCTYPE.

If there is any other way to parse XHTML with jsdom, I would appreciate a tip.

### Basic info:

- **Node.js version:** 16.11.1
- **jsdom version:** 18.0.0

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const options = {
  contentType: 'application/xml'
};
const dom = new JSDOM(`<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><body>&ouml;</body></html>`, options);
```

### How does similar code behave in browsers?

Example code to run in the browser console:
```javascript
new DOMParser().parseFromString(`<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><body>&ouml;</body></html>`, 'application/xml').documentElement.innerHTML;
```

**Comment by lddubeau** [comment]
@Sebmaster That's right. You'd have to set `parser.ENTITIES`. The problem is primarily that saxes does not do any DTD processing, including `DOCTYPE` which is there mainly to load a DTD. There's a `doctype` event that allows you to read the `DOCTYPE` but saxes does nothing with it. 

@cdauth I don't recall any such option, but someone may have changed the code while I was not looking. 😀
